omod-common/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/MainSubResourceController.java

```diff
@@ -75,12 +77,21 @@ public Object retrieve(@PathVariable("resource") String resource, @PathVariable(
 	 */
 	@RequestMapping(value = "/{resource}/{parentUuid}/{subResource}", method = RequestMethod.GET)
 	@ResponseBody
-	public SimpleObject getAll(@PathVariable("resource") String resource, @PathVariable("parentUuid") String parentUuid,
+	public SimpleObject get(@PathVariable("resource") String resource, @PathVariable("parentUuid") String parentUuid,
 	        @PathVariable("subResource") String subResource, HttpServletRequest request, HttpServletResponse response)
 	        throws ResponseException {
 		baseUriSetup.setup(request);
-		RequestContext context = RestUtil.getRequestContext(request, response);
 		SubResource res = (SubResource) restService.getResourceByName(buildResourceName(resource) + "/" + subResource);
+		RequestContext context = RestUtil.getRequestContext(request, response);
+		Converter conv = res instanceof Converter ? (Converter) res : null;

```

Comment:
I've introduced check for Searchable and searching. Using Listable in SubResource makes no sense as SubResource interface defines getAll(parentUuid, context)